### PR TITLE
Update Merge Rules for Intel GPU

### PR DESCRIPTION
**Pull request by EikanWang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #140705

### CODE REVIEW

File: .github/merge_rules.yaml [hunk 1]
```diff
@@ -519,6 +520,14 @@
   - Lint
   - pull
 
+- name: XPU
+  patterns:
+  - '**xpu**'
+  approved_by:
+  - EikanWang
+  - jgong5
+  - gujinghui
```
Comment (EikanWang) [inline .github/merge_rules.yaml:529]: @atalman , it seems like duplicate with `XPU ATen` section. Any side effects?

**Review by zxiiro — APPROVED** [review]

**Comment by EikanWang** [comment]
Merge this PR to #135110 as a single one for efficient PR review.

**Pull request closed without merging**